vcpkg/github-integration.md

```diff
@@ -0,0 +1,94 @@
+
+## The GitHub dependency graph
+
+[!INCLUDE [experimental](../includes/experimental.md)]
+
+> [!TIP]
+> We are actively developing this feature and would like to hear your feedback. Let us know if you have any thoughts about the current functionality or any desired functionality by emailing [vcpkg@microsoft.com](mailto:vcpkg@microsoft.com) or by filing an issue in [vcpkg-docs](https://github.com/microsoft/vcpkg-docs/issues). If you have feedback on any of the GitHub features that this enables, let us know that too and we'll make sure it gets to the right folks at GitHub.
+
+### About the GitHub dependency graph
+
+The GitHub [dependency graph](https://docs.github.com/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph) stores the set of dependencies for a repository. Beyond being able to visualize a repository's dependencies, GitHub builds several useful features on top of this data, including [dependency review](https://docs.github.com/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review) and [Dependabot alerts](https://docs.github.com/code-security/dependabot/dependabot-alerts/about-dependabot-alerts).
+
+### vcpkg integration with the GitHub dependency graph
+
+vcpkg has experimental support for populating the GitHub dependency graph from within a GitHub Actions workflow. To enable this feature, make the following changes in your workflow file:
+
+* Set the `VCPKG_FEATURE_FLAGS` environment variable to include the value `dependencygraph`. 
+* Set the `GITHUB_TOKEN` environment variable to the value `${{ secrets.GITHUB_TOKEN }}`.
+* Give GitHub Actions write access to the contents of your repository by including the following permissions block. This is required to write the dependency graph metadata to your repository. This __will not__ add any commits to your repository or make any other modification to the contents of your repository.
+
+```yaml
+permissions:
+  contents: write
+```
+
+You must enable the GitHub dependency graph in your repository's settings (enabled by default on public repositories). For private repositories, follow the [GitHub documentation to enable the dependency graph](https://docs.github.com/code-security/supply-chain-security/understanding-your-software-supply-chain/configuring-the-dependency-graph#enabling-and-disabling-the-dependency-graph-for-a-private-repository) in order to satisfy this requirement.
+
+### Known limitations
+
+The following are known limitations in the current implementation. __Even though we know of these, if they affect the value for you, we'd still like to hear from you so that we can prioritize what we need to fix and how we should fix it.__
+
+* The version of vcpkg that is bundled with GitHub Actions runners does not have this feature yet. Ensure you are using a version of vcpkg that supports this feature.
```

Comment:
What version is that?  This doc will live on for who knows how long. What is true today won't necessarily be true next month.  It's better to be explicit since this doc may be read in the form years from now.

Reply:
Hmm. It would be nice to rewrite this to

> Ensure you are using at least version NNNN of vcpkg, as the version bundled with GitHub Actions runners may be using an older version.

But, we don't have that version number yet. I think when we do, I can update the doc to be more explicit.

Also, if we have any documentation anywhere that describes how someone could get a new version onto their GitHub Actions runner, that would be useful to link here, but I don't think we have that. If we manage to adopt the `lukka/run-vcpkg` reusable action, that would be a natural place for it. It'll just have to come later.